app.js

```diff
@@ -1,14 +1,40 @@
 var http = require('http');
 var fs = require('fs');
-var path = require('path')
+var path = require('path');
+var mysql = require('mysql');
 const PORT = 3000;
+
+
 http.createServer(function (request, response) {
+    //console.log(request.url);
     
-if (request.url === '/favicon.ico') {           //code to handle favicon.ico image
-    response.writeHead(200, {'Content-Type': 'image/x-icon'} );
-    response.end();
+    if (request.url === '/favicon.ico') {           //code to handle favicon.ico image
+        response.writeHead(200, {'Content-Type': 'image/x-icon'} );
+        response.end();
+        return;
     }
+    var con = mysql.createConnection({
+        host: "localhost",
+        user: "custom",
+        password: "password",
+        database: "calendar_db"
+    });
+
+    if (request.method === 'POST' && request.url === '/') {
+        var body = '';
 
+        request.on('data', function (data) {
+            body =JSON.parse(data);
+            if (body.length > 1e6){
+                request.connection.destroy();
+            }
+            con.query('INSERT INTO dummyTable SET ?' ,body, function (err, result) {
```

Comment:
extract the logic of database update from the logic of request handling by doing something like this: 
* define functions for update, insert delete etc to be more easy
* use your implementation in the request handling logic
This way, we can unit test the database updation logic independently (you can mock requests and everything else and only test whether the data is updated in the database)
In code this would be something like:
```
db_functions.js - new file
const  INSERT_QUERY = 'INSERT INTO dummyTable SET ?';
const UPDATE_QUERY = '...'
...
function addEventToCalendar(con, data, callbackfn) {
  con.query(INSERT_QUERY, data, callbackfn);
}

function updateEventOnCalendar(con, data, callbackfn) {
  con.query(INSERT_QUERY, data, callbackfn);
}
...
```
and so on
and you can use it here by doing:
```
addEventToCalendar(con, body, function () {... })
```
You can refactor this code further by defining the function() {} outside, by doing something like:
```
function dbCallback(err, result) {
  if (err) {
    throw err;
 } 
}
```